node/pkg/processor/observation.go

```diff
@@ -286,30 +286,41 @@ func (p *Processor) handleInboundSignedVAAWithQuorum(ctx context.Context, m *gos
 	//  - enough signatures are present for the VAA to reach quorum
 
 	// Check if we already store this VAA
-	_, err = p.db.GetSignedVAABytes(*db.VaaIDFromVAA(v))
-	if err == nil {
-		p.logger.Debug("ignored SignedVAAWithQuorum message for VAA we already store",
-			zap.String("digest", hash),
-		)
-		return
-	} else if err != db.ErrVAANotFound {
-		p.logger.Error("failed to look up VAA in database",
-			zap.String("digest", hash),
-			zap.Error(err),
-		)
-		return
-	}
+	if v.EmitterChain == vaa.ChainIDPythNet {
+		_, exists := p.getPythNetVAA(v)
+		if exists {
```

Comment:
style nit: the usual convention in go is to combine these checks into a single line

```suggestion
		if _, exists := p.getPythNetVAA(v); exists {
```

See [Effective Go](https://go.dev/doc/effective_go#if) for more details.